File: packages/sdk-snippets/src/targets/node/aws/webhooks/fixtures/empty.js
const { APIGatewayClient, GetApiKeysCommand } = require('@aws-sdk/client-api-gateway');
const readme = require('readmeio');

// Your ReadMe secret; you may want to store this in AWS Secrets Manager
const README_SECRET = 'my-readme-secret';

exports.handler = async event => {
  let statusCode, apiKey, error;

  try {
    const signature = event.headers['ReadMe-Signature'];
    const body = JSON.parse(event.body);
    readme.verifyWebhook(body, signature, README_SECRET);

    const email = body.email;
    const client = new APIGatewayClient();
    const command = new GetApiKeysCommand({ nameQuery: email, includeValues: true });
    const keys = await client.send(command);
    if (keys.items.length > 0) {
      apiKey = keys.items[0].value;
      statusCode = 200;
    } else {
      error = 'Email not found';
      statusCode = 404;
    }
  } catch (e) {
    error = e.message;
    statusCode = error.match(/Signature/) ? 401 : 500;
  }

  return {
    statusCode,
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ apiKey, message: error }),
  };
};

File: packages/sdk-snippets/src/targets/ruby/aws/webhooks/fixtures/empty.rb
require 'aws-sdk-apigateway'
require 'json'
require 'readme/webhook'

# Your ReadMe secret; you may want to store this in AWS Secrets Manager
README_SECRET = "my-readme-secret"

def handler(event:, context:)
    status_code = nil
    api_key = nil
    error = nil

    begin
        # Verify the request is legitimate and came from ReadMe.
        signature = event['headers']['ReadMe-Signature'];
        Readme::Webhook.verify(event['body'], signature, README_SECRET)

        # Look up the API key associated with the user's email address.
        body = JSON.parse(event['body']);
        email = body['email']
        client = Aws::APIGateway::Client.new()
        keys = client.get_api_keys({
            name_query: email,
            include_values: true
        })
        if keys.items.length > 0
            # If multiple API keys are returned for the given email, use the first one.
            api_key = keys.items[0].value
            status_code = 200
        else
            error = "Email not found"
            status_code = 404
        end
    rescue Readme::MissingSignatureError, Readme::ExpiredSignatureError, Readme::InvalidSignatureError => e
        error = e.message
        status_code = 401
    rescue => e
        error = e.message
        status_code = 500
    end

    if status_code == 200
        body = {
            # The user's API key
            apiKey: api_key
        }
    else
        body = {message: error}
    end

    return {
        statusCode: status_code,
        headers: {'Content-Type': 'application/json'},
        body: body.to_json
    }
end